### PR TITLE
Make User extend UserDetails

### PR DESCRIPTION
**Pull request by pikulo-kama**

### CODE REVIEW

File: src/main/java/com/arthurdrabazha/openmind/model/User.java [hunk 1]
```diff
@@ -70,8 +85,36 @@
...
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return isEnabled;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return false;
+    }
```
Comment (MarrMika) [inline src/main/java/com/arthurdrabazha/openmind/model/User.java:114]: I do believe that by default we need to have true value here for the future.

File: src/main/java/com/arthurdrabazha/openmind/model/User.java [hunk 2]
```diff
@@ -51,6 +61,11 @@
 
     private Long dislikes;
 
+    @Enumerated(EnumType.STRING)
+    private UserRole role;
+
+    private Boolean isEnabled;
+
```
Comment (MarrMika) [inline src/main/java/com/arthurdrabazha/openmind/model/User.java:68]: How are you going to identify that user is active? What is the difference in usage between isEnabled and nonActiveDaysBeforeDelete ?

**Review by MarrMika — APPROVED** [review]
Good! Just fix one comment and can merge.

File: src/main/java/com/arthurdrabazha/openmind/model/User.java (hunk 2 above)
Comment (pikulo-kama) [reply src/main/java/com/arthurdrabazha/openmind/model/User.java:68]: Actually idea was to use _isEnabled_ for admins to block users, and _nonActiveDaysBeforeDelete_ for scheduled operations - such as remove account after time have settled by the user have passed

Comment (MarrMika) [reply src/main/java/com/arthurdrabazha/openmind/model/User.java:68]: Under what conditions do we consider a user to be blocked?

Comment (pikulo-kama) [reply src/main/java/com/arthurdrabazha/openmind/model/User.java:68]: Here is the case - person posted some post with sensitive information for some nation or group of people.
In that case admin or some moderator can block this user

Comment (MarrMika) [reply src/main/java/com/arthurdrabazha/openmind/model/User.java:68]: I've got you, let it be. Thanks!

**Pull request merged**